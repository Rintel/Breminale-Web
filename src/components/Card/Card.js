import React from 'react';
import PropTypes from 'prop-types';
import { GridTile } from 'material-ui/GridList';
import IconButton from 'material-ui/IconButton';
import StarBorder from 'material-ui/svg-icons/toggle/star-border';
import { Link } from 'react-router-dom';


const propTypes = {
    event: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    location_id: PropTypes.number.isRequired,
    image_url: PropTypes.string.isRequired,
    image_url_medium: PropTypes.string.isRequired,
    image_url_small: PropTypes.string.isRequired,
    start_time: PropTypes.string.isRequired,
    soundcloud_url: PropTypes.string.isRequired,
    soundcloud_user_id: PropTypes.string.isRequired,
  }).isRequired
};

const EventCard = (props) => {
  const { event } = props;

  return (
    <Link to={`/events/${event.id}`}>
      <GridTile
        key={event.id}
        title={event.name}
        subtitle={<span>by <b>{event.description}</b></span>}
        actionIcon={<IconButton><StarBorder color="white" /></IconButton>}
      >
        <img src={"https://via.placeholder.com/350x150"} />
      </GridTile>
    </Link>
  );
};

EventCard.propTypes = propTypes;

export default EventCard;
