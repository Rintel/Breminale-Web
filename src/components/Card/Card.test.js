import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';


const baseProps = {
  event:{
    id: 3,
    name: 'Event',
    description: 'Description',
    location_id: 123,
    image_url: 'url',
    image_url_medium: 'url',
    image_url_small: 'url',
    start_time: 'time',
    soundcloud_url: 'url',
    soundcloud_user_id: 'id'
  }
};

describe('Card Component is rendering', () => {
  test('valid props are passed', () => {
    const wrapper = shallow(
      <Card {...baseProps} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
