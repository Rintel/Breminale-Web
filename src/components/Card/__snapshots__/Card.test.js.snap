// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page rendering valid props are passed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventCard
    event={
      Object {
        "description": "Description",
        "id": 3,
        "image_url": "url",
        "image_url_medium": "url",
        "image_url_small": "url",
        "location_id": 123,
        "name": "Event",
        "soundcloud_url": "url",
        "soundcloud_user_id": "id",
        "start_time": "time",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <GridTile
        actionIcon={
          <IconButton
            disableTouchRipple={false}
            disabled={false}
            iconStyle={Object {}}
            tooltipPosition="bottom-center"
            touch={false}
          >
            <ToggleStarBorder
              color="white"
            />
          </IconButton>
        }
        actionPosition="right"
        cols={1}
        containerElement="div"
        rows={1}
        subtitle={
          <span>
            by 
            <b>
              Description
            </b>
          </span>
        }
        title="Event"
        titleBackground="rgba(0, 0, 0, 0.4)"
        titlePosition="bottom"
      >
        <img
          src="https://via.placeholder.com/350x150"
        />
      </GridTile>,
      "replace": false,
      "to": "/events/3",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": "3",
      "nodeType": "class",
      "props": Object {
        "actionIcon": <IconButton
          disableTouchRipple={false}
          disabled={false}
          iconStyle={Object {}}
          tooltipPosition="bottom-center"
          touch={false}
        >
          <ToggleStarBorder
            color="white"
          />
        </IconButton>,
        "actionPosition": "right",
        "children": <img
          src="https://via.placeholder.com/350x150"
        />,
        "cols": 1,
        "containerElement": "div",
        "rows": 1,
        "subtitle": <span>
          by 
          <b>
            Description
          </b>
        </span>,
        "title": "Event",
        "titleBackground": "rgba(0, 0, 0, 0.4)",
        "titlePosition": "bottom",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "src": "https://via.placeholder.com/350x150",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <GridTile
          actionIcon={
            <IconButton
              disableTouchRipple={false}
              disabled={false}
              iconStyle={Object {}}
              tooltipPosition="bottom-center"
              touch={false}
            >
              <ToggleStarBorder
                color="white"
              />
            </IconButton>
          }
          actionPosition="right"
          cols={1}
          containerElement="div"
          rows={1}
          subtitle={
            <span>
              by 
              <b>
                Description
              </b>
            </span>
          }
          title="Event"
          titleBackground="rgba(0, 0, 0, 0.4)"
          titlePosition="bottom"
        >
          <img
            src="https://via.placeholder.com/350x150"
          />
        </GridTile>,
        "replace": false,
        "to": "/events/3",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": "3",
        "nodeType": "class",
        "props": Object {
          "actionIcon": <IconButton
            disableTouchRipple={false}
            disabled={false}
            iconStyle={Object {}}
            tooltipPosition="bottom-center"
            touch={false}
          >
            <ToggleStarBorder
              color="white"
            />
          </IconButton>,
          "actionPosition": "right",
          "children": <img
            src="https://via.placeholder.com/350x150"
          />,
          "cols": 1,
          "containerElement": "div",
          "rows": 1,
          "subtitle": <span>
            by 
            <b>
              Description
            </b>
          </span>,
          "title": "Event",
          "titleBackground": "rgba(0, 0, 0, 0.4)",
          "titlePosition": "bottom",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "src": "https://via.placeholder.com/350x150",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
